apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2015,6 +2058,28 @@ check_withdraw_error_msg(Msg, D) ->
 check_update_err_msg(Msg, D) ->
     check_op_error_msg(?UPDATE_ERR, Msg, D).
 
+check_shutdown_err_msg(#{ channel_id := ChanId } = Msg,
+                       #data{ on_chain_id = ChanId
+                            , op = Op } = D) ->
+    case Op of
+        #op_close{data = #op_data{signed_tx = SignedTx}} ->
+            TxHash = aetx_sign:hash(SignedTx),
+            case aec_chain:find_tx_location(TxHash) of
+                Block when is_binary(Block) ->
+                    %% Tx is already on chain - ignore error msg
+                    {error, tx_already_on_chain};
+                mempool ->
+                    %% For now, treat it as if it's on chain
+                    {error, tx_already_on_chain};
+                _ ->
```

Comment:
```suggestion
                not_found->
                    {error, tx_not_found};
                none ->
```

I think it is worth differentiating between the heads: `aec_chain:find_tx_location/1` can return also `not_found|none` with different semantics:
* `not_found` means that the node has not yet seen this transaction. In this case either our node had not yet synced the posted transaction or it wasn't posted on-chain in the first place
* `none` means it was rejected as an invalid one (for ex. `nonce` didn't match). I think in this particular case it is OK to rollback to `open` state

Reply:
So if it wasn't posted on on-chain in the first place, why would we reject the error message?